### PR TITLE
[processor] Composition order

### PR DESCRIPTION
**Pull request by plesiecki**
## Description

One more test case for composition order. 
An expected order would be `mc_folder2` before `mc_foo`.

Follow-up of https://github.com/tivac/modular-css/pull/890

## Motivation and Context

Consistent composition map. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tivac** [comment]
Sorry, missed the notification on this one @plesiecki, I'll try to get some eyes on it tonight ❤

**Comment by plesiecki** [comment]
Hello, just to let you know that with 50220c1b80c41c8757c56e3727bd718b95a0370b I see consistent results. Maybe it's worth reconsidering.

**Comment by plesiecki** [comment]
I've added few more tests for composition

**Comment by plesiecki** [comment]
Looks like a removing this line below can help
https://github.com/tivac/modular-css/blob/e64362e087f4ae6113a1b622e59dab827a9f2e97/packages/processor/processor.js#L513

**Comment by plesiecki** [comment]
`PASS packages/processor/test/composition.test.js`; rest of the snapshots likely should be regenerated.

**Comment by tivac** [comment]
The problem with https://github.com/tivac/modular-css/commit/50220c1b80c41c8757c56e3727bd718b95a0370b was, as I recall, that it outputs the composition selectors in the wrong order. Since we're already building the dependency graph including all the selectors, I think it probably needs to start generating composition data from that instead of just shoving it onto an array?

I haven't looked at this for a bit despite saying I would, I'm sorry.

File: packages/processor/test/specimens/composes/external-compose-via-local-compose.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+.ruleWithLocalComposes {
+    composes: ruleWithSingleCompose;
+    composes: blue from "../deps/classes.css";
+}
+
+.nextRuleWithLocalComposes {
+    composes: ruleWithSingleCompose;
+}
+
+.nextRuleWithLocalComposes {
```
Comment (tivac) [inline packages/processor/test/specimens/composes/external-compose-via-local-compose.css:14]: is the duplication of `.nextRuleWithLocalComposes` intended here?

File: packages/processor/test/specimens/composes/external-compose-via-local-compose3.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.ruleWithSingleCompose {
+    composes: red from "../deps/classes.css";
+}
+
+.ruleWithLocalComposes {
+    composes: ruleWithSingleCompose;
+}
+
+.ruleWithLocalComposes {
+    composes: blue from "../deps/classes.css";
```
Comment (tivac) [inline packages/processor/test/specimens/composes/external-compose-via-local-compose3.css:10]: Duplication of `.ruleWithLocalComposes` intended?

File: packages/processor/test/specimens/composes/external-compose-via-local-compose4.css [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.ruleWithSingleCompose {
+    composes: red from "../deps/classes.css";
+}
+
+.ruleWithLocalComposes {
+    composes: blue from "../deps/classes.css";
+}
+
+.ruleWithLocalComposes {
```
Comment (tivac) [inline packages/processor/test/specimens/composes/external-compose-via-local-compose4.css:9]: Same question here.

**Comment by plesiecki** [comment]
> The problem with [50220c1](https://github.com/tivac/modular-css/commit/50220c1b80c41c8757c56e3727bd718b95a0370b) was, as I recall, that it outputs the composition selectors in the wrong order. Since we're already building the dependency graph including all the selectors, I think it probably needs to start generating composition data from that instead of just shoving it onto an array?
> 
> I haven't looked at this for a bit despite saying I would, I'm sorry.

No problem. Forget about https://github.com/tivac/modular-css/commit/50220c1b80c41c8757c56e3727bd718b95a0370b, I've approached it differently — https://github.com/tivac/modular-css/pull/892/commits/43c126a57ca34f7d064dd803e32d204999199502

File: packages/processor/test/specimens/composes/external-compose-via-local-compose.css (hunk 1 above)
Comment (plesiecki) [reply packages/processor/test/specimens/composes/external-compose-via-local-compose.css:14]: It's not necessarily a duplication, but just a continuation of class definition, an alternative notation for:
```css
.nextRuleWithLocalComposes {
    composes: ruleWithSingleCompose;
    composes: blue from "../deps/classes.css";
}
```
I just want to make sure that every case works. 😅

File: packages/processor/test/specimens/composes/external-compose-via-local-compose3.css (hunk 2 above)
Comment (plesiecki) [reply packages/processor/test/specimens/composes/external-compose-via-local-compose3.css:10]: https://github.com/tivac/modular-css/pull/892#discussion_r991872610

File: packages/processor/test/specimens/composes/external-compose-via-local-compose4.css (hunk 3 above)
Comment (plesiecki) [reply packages/processor/test/specimens/composes/external-compose-via-local-compose4.css:9]: https://github.com/tivac/modular-css/pull/892#discussion_r991872610

**Comment by tivac** [comment]
> No problem. Forget about [50220c1](https://github.com/tivac/modular-css/commit/50220c1b80c41c8757c56e3727bd718b95a0370b), I've approached it differently — [43c126a](https://github.com/tivac/modular-css/commit/43c126a57ca34f7d064dd803e32d204999199502) and I'd like to know your opinion on this.

I need to spend some time really understanding why the fix is to remove a dependency that, at first glance, seems to be correct. The snapshots I spot-checked all looked right so I'm confident it's doing the right thing, I just need to know **why**. I haven't forgotten about this, just need to claw some time back to poke at it further to make sure I understand.

File: packages/processor/processor.js [hunk 4]
```diff
@@ -508,9 +508,7 @@ class Processor {
         this._addSelector(name, selector);
 
         refs.forEach(({ name : depSelector }) => {
-            const depSelectorId = this._addSelector(dep, depSelector);
-
-            graph.addDependency(selectorId, depSelectorId);
```
Comment (plesiecki) [inline packages/processor/processor.js:513]: It works thanks to that piece of code https://github.com/tivac/modular-css/blob/e026d812b17526a8f7e756205f5ed7424563dfee/packages/processor/plugins/composition.js#L77-L79

**Comment by tivac** [comment]
Thanks again for tracking this down @plesiecki, I finally think I have a handle on *why* the fix works and this is released as [`@modular-css/processor@28.1.5`](https://github.com/tivac/modular-css/releases/tag/%40modular-css%2Fprocessor%4028.1.5).

**Pull request merged**